From 8bbccf7263fb6bd902baef5c15e9472b58b79fd7 Mon Sep 17 00:00:00 2001
From: "neal.yen" <neal.yen@mediatek.com>
Date: Tue, 28 Oct 2025 10:57:10 +0800
Subject: [PATCH] net: phy: sfp: add shared capability to mod-def0 gpio

The combo SFP/PHY dynamic switching through the passive mux will use the
mod-def0 GPIO to detect whether the SFP module has been inserted.
We need to change this GPIO to shared to avoid failures in GPIO
requests within mtk_add_mux().

Signed-off-by: neal.yen <neal.yen@mediatek.com>

diff --git a/drivers/net/phy/sfp.c b/drivers/net/phy/sfp.c
index 5f4dc0a..43bcc16 100644
--- a/drivers/net/phy/sfp.c
+++ b/drivers/net/phy/sfp.c
@@ -156,7 +156,7 @@ static const char *gpio_names[] = {
 };
 
 static const enum gpiod_flags gpio_flags[] = {
-	GPIOD_IN,
+	GPIOD_IN | GPIOD_FLAGS_BIT_NONEXCLUSIVE,
 	GPIOD_IN,
 	GPIOD_IN,
 	GPIOD_ASIS,
@@ -597,7 +597,7 @@ static unsigned int sfp_gpio_get_state(struct sfp *sfp)
 	unsigned int i, state, v;
 
 	for (i = state = 0; i < GPIO_MAX; i++) {
-		if (gpio_flags[i] != GPIOD_IN || !sfp->gpio[i])
+		if ((gpio_flags[i] & GPIOD_IN) != GPIOD_IN || !sfp->gpio[i])
 			continue;
 
 		v = gpiod_get_value_cansleep(sfp->gpio[i]);
@@ -3095,7 +3095,7 @@ static int sfp_probe(struct platform_device *pdev)
 	}
 
 	for (i = 0; i < GPIO_MAX; i++) {
-		if (gpio_flags[i] != GPIOD_IN || !sfp->gpio[i])
+		if ((gpio_flags[i] & GPIOD_IN) != GPIOD_IN || !sfp->gpio[i])
 			continue;
 
 		sfp->gpio_irq[i] = gpiod_to_irq(sfp->gpio[i]);
-- 
2.45.2

